std/typecons.d

```diff
@@ -2185,6 +2185,69 @@ Params:
         _isNull = false;
     }
 
+    /**
+      If they are both null, then they are equal. If one is null and the other
+      is not, then they are not equal. If they are both non-null, then they are
+      equal if their values are equal.
+      */
+    bool opEquals()(auto ref inout(typeof(this)) rhs) inout
+    {
+        if(_isNull)
+            return rhs._isNull;
+        if(rhs._isNull)
+            return false;
+        return _value == rhs._value;
```

Comment:
It _is_ private. And since this is a member function, it's accessible. The fact that it's from another instance doesn't matter. The normal thing to do in a function like `opEquals` or `opCmp` is to use the member variables directly, and it's often the case that that's the only way that they can be used, because they're private. It would be incredibly bizarre to call a member function to access a member to compare it in `opEquals` rather than just accessing it directly.